### PR TITLE
Refactor Starknet test setup

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- None, hopefully

## Development related changes

- Replace the multiple `fn setup(acc_balance: Option<u128>) -> (Starknet, ContractAddress, ...)` definitions with:
  - `fn setup_starknet_with_unvalidated_account(acc_balance: u128)`
  - `fn setup_starknet_with_unvalidated_account_and_state_capacity(acc_balance: u128, state_archive: StateArchiveCapacity)`
- Instead of passing around the fee token addresses, rely on `starknet.block_context.chain_info()`
- Created while implementing the proposal in https://github.com/0xSpaceShard/starknet-devnet-rs/issues/272#issuecomment-1924033491
  - With this PR it should be easier
- Extract `(FlattenedSierraClass, FieldElement)` to `SierraWithHash`

## Checklist:

- [x] Checked out the [contribution guidelines](CONTRIBUTING.md)
- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] No spelling errors - `./scripts/check_spelling.sh`
- [x] Performed code self-review
- [x] Rebased to the latest commit of the target branch (or merged it into my branch)
- [x] Updated the docs if needed - `./website/README.md`
- [x] Linked the [issues](https://github.com/0xSpaceShard/starknet-devnet-rs/issues) resolvable by this PR - [linking info](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] Updated the tests if needed; all passing - [execution info](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#test-execution)

### CODE REVIEW

File: crates/starknet-devnet-core/src/starknet/mod.rs [hunk 1]
```diff
@@ -1397,11 +1399,58 @@ mod tests {
...
     use crate::state::state_diff::StateDiff;
-    use crate::traits::{Accounted, HashIdentified};
+    use crate::traits::{Accounted, Deployed, HashIdentified};
     use crate::utils::test_utils::{
-        dummy_contract_address, dummy_declare_transaction_v1, dummy_felt,
+        cairo_0_account_without_validations, dummy_contract_address, dummy_declare_transaction_v1,
+        dummy_felt,
     };
 
+    /// Initializes starknet with 1 account that doesn't require tx without validation and with the
```
Comment (marioiordanov) [inline crates/starknet-devnet-core/src/starknet/mod.rs:1408]: This comment is kind of misleading to me. 
```suggestion
    /// Initializes starknet with 1 account that doesn't perform actual signature validation on transactions and with the specified state archive capacity
```

File: crates/starknet-devnet-core/src/starknet/mod.rs [hunk 2]
```diff
@@ -1397,11 +1399,58 @@ mod tests {
...
+    /// specified state archive capacity
+    pub(crate) fn setup_starknet_with_unvalidated_account_and_state_capacity(
```
Comment (marioiordanov) [inline crates/starknet-devnet-core/src/starknet/mod.rs:1410]: unvalidated_account doesn't reflect the real property of the account. I can propose a longer name setup_starknet_with_no_signature_check_account_and_state_capacity

**Review by marioiordanov — CHANGES_REQUESTED** [review]

File: crates/starknet-devnet-core/src/starknet/mod.rs (hunk 2 above)
Comment (FabijanC) [reply crates/starknet-devnet-core/src/starknet/mod.rs]: Fixed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/494/commits/896b8cf81536b0dce9fae2d5a9fc95c7b037a9a3

File: crates/starknet-devnet-core/src/starknet/mod.rs (hunk 1 above)
Comment (FabijanC) [reply crates/starknet-devnet-core/src/starknet/mod.rs]: Fixed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/494/commits/896b8cf81536b0dce9fae2d5a9fc95c7b037a9a3

**Review by marioiordanov — APPROVED** [review]

**Pull request merged**